### PR TITLE
Add checksum verification on FITS file read

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request introduces the last element for fixity metadata support. The `checksum` argument is now passed to read methods. 
For now it is only implemented on `Observation`, but this will be added to `Dataset` and `FluxMap`. 

**Dear reviewer**

### CODE REVIEW

**Review by bkhelifi — DISMISSED** [review]
Thanks @registerrier . It looks good... I have just left a minor comment about a docstring

File: gammapy/data/gti.py [hunk 1]
```diff
@@ -130,21 +131,31 @@ def create(cls, start, stop, reference_time=None):
...
-    def read(cls, filename, hdu="GTI", format="gadf"):
+    def read(cls, filename, hdu="GTI", format="gadf", checksum=False):
         """Read from FITS file.
 
         Parameters
         ----------
-        filename : `pathlib.Path` or str
-            Filename.
-        hdu : str, optional
+        filename : `pathlib.Path`, str
```
[the remark this answers is not in the document]
Comment (registerrier) [reply gammapy/data/gti.py:139]: Hmm, I don't remember changing this...

**Comment by registerrier** [comment]
Maybe a general question, here. 
There are basically two options to perform the checksum verification:
- `fits.open(filename, checksum=checksum)` will check ALL hdus and emit `AstropyUserWarning` if they are inconsistent
- `hdu.verify_checksum()` will return 1 if the verification worked. A `UserWarning` can be raised for the explicit hdu tested
Functions using `Table.read()` do not allow a check unless opening specifically the correct hdu. For now I followed option 2 for these. See e.g. `Observation.read()`.
This approach can become tedious when there are many hdus (but it can be hidden in an helper function). In this case, using  the first option seems easier and more natural. This is what is used in e.g. `MapDataset.read`.
This creates an inconsistent approach which is not very nice.

What do you think @adonath @bkhelifi @AtreyeeS @QRemy ? Is it fine with hetegeneous behaviour or should we stick to one. e.g. the second one, even if it is not very elegant to have repeated warnings in the file is opened several times to access different hdus?

**Comment by bkhelifi** [comment]
Knowing exactly which hdu has been modified is very useful. Is the function `fits.open(filename, checksum=checksum)' precises which HDU is in fault in case of unsuccessful check? If not, a check for each hdu might be more informative...

**Comment by bkhelifi** [comment]
> > Knowing exactly which hdu has been modified is very useful. Is the function `fits.open(filename, checksum=checksum)' precises which HDU is in fault in case of unsuccessful check? If not, a check for each hdu might be more informative...
> 
> `fits.open` will emit one warning per faulty HDU. All informations are available. It is mostly less efficient and a bit too verbose because it performs check for all HDUs even if one accesses only one.

I understand then your choice, which seems reasonable then. Having a different behaviour between the DL3 and the DL4 does not look odd for me, as the data are different. The clarity of information given to the user is an important driver of the choice.

File: gammapy/datasets/tests/test_spectrum.py [hunk 2]
```diff
@@ -593,11 +594,24 @@ def test_to_ogip_files_checksum(self, tmp_path):
         dataset = self.dataset.copy(name="test")
         dataset.write(tmp_path / "test.fits", format="ogip", checksum=True)
 
-        for file in ["test.fits", "test_arf.fits", "test_rmf.fits", "test_bkg.fits"]:
-            hdul = fits.open(tmp_path / file)
-            for hdu in hdul:
-                assert "CHECKSUM" in hdu.header
-                assert "DATASUM" in hdu.header
+        for name in ["test.fits", "test_arf.fits", "test_rmf.fits", "test_bkg.fits"]:
+            # TODO: this shouldnot emit AstropyUserWarning
```
Comment (bkhelifi) [inline gammapy/datasets/tests/test_spectrum.py:598]:
```suggestion
            # TODO: this should not emit AstropyUserWarning
```

**Comment by registerrier** [comment]
Merging now.

**Pull request merged**